### PR TITLE
feat: FE Init + base layout

### LINKED ISSUE #46: Phase [7] FE Setup & Layout
---
name: Feature
about: Neues Feature (UI/BE/Build)
labels: ["feature"]
---

## Kontext
Erstellung der Basisstruktur für das Frontend-MVP.  
Das Ziel ist ein funktionierendes Next.js-Projekt mit shadcn/ui-Komponenten, App Router und CI-Anbindung.  
Keine Business-Logik – rein UI-Skeleton für die spätere Anzeige.

## Ziel / Scope
- Next.js App mit App Router-Struktur
- shadcn/ui integriert (Designsystem)
- Header, Sidebar und Content-Bereich
- Navigation zwischen Kunden, Profilen, Rechnungen

## Akzeptanzkriterien / DoD
- [x] Projektstruktur erstellt (`frontend/`)
- [x] shadcn/ui funktioniert mit Tailwind
- [x] Seitenrouten `/customers`, `/profiles`, `/invoices` vorhanden
- [x] Navigation zwischen den Seiten über Sidebar
- [ ] CI/CD Frontend-Test läuft grün

## Tasks
- [x] Next.js + shadcn/ui Setup
- [x] Tailwind konfigurieren (inkl. base/theme)
- [x] Layout-Komponenten (Header, Sidebar, Content)
- [x] CI-Step in `.github/workflows/ci.yml` prüfen

## Risiken / Out-of-Scope
- Keine Datenanbindung
- Kein State-Management
- Kein Styling-Finetuning

## Referenzen (Docs)
- Next.js App Router: https://nextjs.org/docs/app  
- shadcn/ui: https://ui.shadcn.com  
- FastAPI: https://fastapi.tiangolo.com

### PR DESCRIPTION
**Pull request by wontknow**
## Zusammenfassung
(Was wurde gemacht? Warum?)

## Verwandte Issues
Closes #46 

## Änderungen
- FE Init
- Basis Layout Dashboard,  Kunden, Profile, Rechnungen

## Screenshots / Logs
image

## Checkliste (DoD)
- [ ] Funktionalität erfüllt (Akzeptanzkriterien)
- [ ] Readme/Docs aktualisiert (falls nötig)
- [ ] Keine Secrets/Keys committed
- [ ] Lint/Build grün
- [ ] Manuelle Tests (happy path & edge cases)
- [ ] CORS geprüft (Dev/Prod)  
- [ ] PDF erzeugt & visuell geprüft  
- [ ] (optional) E-Rechnung Validierung dokumentiert

### CODE REVIEW

File: .github/workflows/frontend-ci.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        uses: actions/checkout@v4
+
+      # Installiere pnpm (BEVOR setup-node den Cache nutzt)
+      - name: Setup pnpm
+        uses: pnpm/action-setup@v4
+        with:
+          version: 10.20.0 # gleiche Version wie lokal
+
+      # Node + Cache aktivieren
+      - name: Setup Node.js
```
[the remark this answers is not in the document]
Comment (wontknow) [reply .github/workflows/frontend-ci.yml]: @copilot open a new pull request to apply changes based on [this feedback](https://github.com/wontknow/Billino/pull/62#discussion_r2484826006)
